### ISSUE TITLE
Record stats on fitting solution run?

### ISSUE THREAD

**Issue by TorkelE** [sebapersson/PEtab.jl#163]
E.g. DifferentialEquations record various data of the simulation (like the number f function evaluations):
![image](https://github.com/sebapersson/PEtab.jl/assets/18099310/c6c8789e-ec0c-4dde-93da-8ce80a099766)
While not directly needed, it can be useful for debugging purposes, or analysis of performance, and similar.

Stuff which it could be useful to know:
- Number of function evaluations (of the cost function).
- Number of occurrences of various warnings/simulations problems (e.g. maxiters reached, instability detected, and similar).

Another advantage of keeping this is that, rather than notifying the user of problems by displaying DiffEq warning messages, one could provide a summary at the end (auto display of diffeq warnings could still be an option). E.g. *if* the number of warnings of a certain type exceeds some number (e.g. 0), then at the end a summary is printed:
```
Warning: A large number of simulations encountered problems
Number of model simulations 1000
Number of maxiters reached: 96
```

**Comment by TorkelE** [comment]
Yes, something like that sounds good. A little bit unsure about the state reset function? Do You mean for resetting these statistics or something else? Wouldn't these statistics be stored in the optimisation output structure (and whatever thing keeps track of the optimisation during the optimisation process)?

**Comment by sebapersson** [comment]
You are correct (somehow I imagined they should be in the `PEtabODEProblem`), but in the optmisation ouput structure makes far more sense!

**Comment by sebapersson** [comment]
I have changed the milestone to v4, as this would be great to have, but I do not want to hold back PEtab v3.
